django_emarsys/managers.py

```diff
@@ -2,81 +2,94 @@
 
 from __future__ import unicode_literals
 
-import sys
+import logging
 
+from django.conf import settings
 from django.db import models
+from django.utils.html import conditional_escape
 
 from emarsys import EmarsysError
 
-from . import api
+from django_emarsys import api, context_provider_registry
+from django_emarsys.exceptions import UnknownEventNameError
 
+log = logging.getLogger(__file__)
 
-class EventManager(models.Manager):
+
+class EmarsysEventIdManager(models.Manager):
     def sync_events(self):
-        """
-        Sync locally available events, identified by their name, with emarsys.
-
-        In particular:
-            * Add events that don't exist locally.
-            * Add missing emarsys event ids.
-            * Disable events that don't exist on emarsys by setting their
-              emarsys_id to None. Events are not deleted so as to not lose
-              associated EventData information.
-
-        Events are identified by their name. This allows adding and
-        implementing events in advance without knowing their emarsys id.
-        Syncing will then add the emarsys id to the events so that they can be
-        triggered.
-        """
-        num_new_events = 0
-        num_added_ids = 0
-        num_updated_ids = 0
-        num_deleted_ids = 0
+        emarsys_events = api.get_events()
+        for name, emarsys_id in emarsys_events.items():
+            obj, created = self.get_or_create(
+                event_name=name,
+                defaults={'emarsys_id': emarsys_id})
+            if not created:
+                obj.emarsys_id = emarsys_id
+                obj.save()
+
+
+class NewEventManager(models.Manager):
+    def trigger(self, event_name, recipient_email, user=None, data=None,
+                source='automatic', async=False):
+        if async:
+            raise NotImplementedError
```

Comment:
Ok, I'll take it out for now, but I think it's a very good and important idea to do this asynchronously... especially if it might need to retrieve the ID now and then (which in practice is ultra rare, of course).